content/docs/iac/using-pulumi/iac-least-privileges/_index.md

```diff
@@ -0,0 +1,181 @@
+
+```bash
+pulumi stack init --teams YourTeamName
+```
+
+This grants the specified team immediate read/write access to the new stack.
+
+### 3. Secure Deployment Approaches for Production
+
+Choose one of these secure deployment approaches in your sensitive environments:
+
+### Option A: Pulumi Deployments (with GitHub and Other Integrations)
+
+[Pulumi Deployments](/docs/pulumi-cloud/deployments/) provides automated, managed, and secure infrastructure deployments:
+
+- **Automated GitHub integration:**  
+  Automatically run `pulumi preview` on PRs and `pulumi up` upon PR merge.
+- **REST API:**  
+  Trigger deployments programmatically from your custom workflows or third-party CI/CD systems using the Pulumi Deployments REST API.
+
+**Setup steps for GitHub integration:**
+
+1. **Install the Pulumi GitHub App:**
+   - Navigate to Pulumi Cloud Console → `Settings > Integrations`.
+   - Install the app and authorize it for your GitHub repository.
+
+2. **Configure deployment triggers:**
+   - Navigate to your stack in Pulumi Cloud Console → `Stack Settings > Deploy`.
+   - Set deployment triggers (e.g., PR merges to `main` branch).
+
+**Learn more:** [Pulumi Deployments GitHub Integration](https://www.pulumi.com/docs/iac/using-pulumi/continuous-delivery/github-app/)
+
+**REST API documentation:** [Pulumi Deployments REST API](https://www.pulumi.com/docs/pulumi-cloud/deployments/rest-api/)
+
+---
+
+### Option B: CI/CD with GitHub Actions and OIDC Authentication
+
+Use GitHub Actions with Pulumi’s OIDC integration for secure, token-less deployments:
+
+- Configure Pulumi Cloud to trust GitHub’s OIDC provider.
+- Obtain a Pulumi access token scoped specifically to a team rather than the entire organization.
+
+**Full Example Workflow (Team-Scoped):**
+
+```yaml
+name: Pulumi Deployment
+
+on:
+  push:
+    branches:
+      - main
+
+permissions:
+  id-token: write
+  contents: read
+
+jobs:
+  deploy:
+    runs-on: ubuntu-latest
+
+    steps:
+      - name: Checkout Repository
+        uses: actions/checkout@v4
+
+      - name: Authenticate with Pulumi (Team-scoped)
+        uses: pulumi/auth-actions@v1
+        with:
+          organization: your-org-name
+          requested-token-type: urn:pulumi:token-type:access_token:team
+          team: your-team-name
+
+      - name: Deploy Infrastructure
+        uses: pulumi/actions@v6
+        with:
+          command: up
+          stack-name: your-org-name/your-stack-name
+```
+
+- `pulumi/auth-actions` exchanges an OIDC token for a team-scoped Pulumi access token.
+- `pulumi/actions` executes `pulumi up` adhering to the assigned team permissions.
+
+**Detailed documentation:**  
+[Pulumi GitHub Actions and OIDC Integration](https://www.pulumi.com/docs/iac/using-pulumi/continuous-delivery/github-actions/)
+
+---
+
+### Option C: CI/CD with Other Providers
+
+Pulumi integrates seamlessly with many other popular CI/CD platforms beyond GitHub Actions, such as:
```

Comment:
should we add a paragraph around subject customization to scope access on the provider side?